core/blockchain.go

```diff
@@ -1827,14 +1835,98 @@ func (bc *BlockChain) insertSideChain(block *types.Block, it *insertIterator) (i
 	}
 	if len(blocks) > 0 {
 		log.Info("Importing sidechain segment", "start", blocks[0].NumberU64(), "end", blocks[len(blocks)-1].NumberU64())
-		return bc.insertChain(blocks, false)
+		return bc.insertChain(blocks, false, false)
 	}
 	return 0, nil
 }
 
+// recoverAncestors finds the closest ancestor with available state and re-execute
+// all the ancestor blocks since that.
+// recoverAncestors is only used in catalyst mode.
```

Comment:
I think we need to get rid of  the notion of "catalyst mode" in chain processing.